### ISSUE TITLE
3D support

### ISSUE THREAD

**Issue by SebDBI** [MIAnalyzer/MIA#5]
This looks like a very useful software! Are you planning support for 3D images?

**Comment by nkoerb** [comment]
Hey, 
in principle z-stacks (3D) are supported. There are 2 options for working with them, having a mask/prediction for every layer in the stack or having a prediction for the whole stack. 
What is not supported are 3D models (i.e. using 3D convolutions), which could be rather easliy implemented.

**Comment by SebDBI** [comment]
So, it is possible to annotate 3D images and train a model from these annotations but the model is only processing slice by slice, is that so?
Are 3D models on your roadmap? I was also looking for 3D annotations tools, do you plan anything for this?

**Comment by nkoerb** [comment]
Hey, 
so currently there are 2 option:
1) each slide of a 3D stack is processed independently, resulting in a prediction for each slide. The model only processes the information of a single slide at a time.
2) A 3D stack can be used as input and generate a joint prediction for the whole stack. The model processes information from all slides and gives a prediction based on that. The model still uses 2D kernels, which means that no 3D features can be detected (but 2D features from different slides) and the resulting output is 2D.

option 3 which is not implemented would be to use 3D Kernels that can detect 3D information and produces a 3D output.

Basically, as 1) and 2) already exist it is not very much work to implement. Currently option 3 not on the immediate roadmap, probably if more poeple request it or I have a little time left. Sorry.

**Comment by SebDBI** [comment]
Thank you for the details. It might actually be sufficient for many 3D problems to operate slice by slice.

An important feature though is then to support sparse annotations when training from 3D stacks as fully annotating 3D images can be very time consuming. For this, it would also be very useful that the tools provided to annotate bring some  3D supporr (e.g. 3D wand and interpolation between 2D contours).
